component-monitor/src/main.rs

```
@@ -0,0 +1,238 @@
+                let path = e.path();
+                if path.is_dir() {
+                    match fs::remove_dir_all(&path) {
+                        Ok(()) => {
+                            debug!(target: "files", "removed {0:?} recursively", &path)
+                        }
+                        Err(e) => {
+                            error!(target: "files", "failed to remove {0:?}: {e}", &path)
+                        }
+                    }
+                } else {
+                    match fs::remove_file(&path) {
+                        Ok(()) => {
+                            debug!(target: "files", "removed {0:?}", &path)
+                        }
+                        Err(e) => {
+                            error!(target: "files", "failed to remove {0:?}: {e}", &path)
+                        }
+                    }
+                }
+            }
+            Err(e) => {
+                error!(target: "files", "{e}");
+            }
+        }
+    }
+}
+
+/// Check if the sentinel file is current, else clean the target directory and populate again,
+/// sentinel file being last
+fn populate(source: &PathBuf, sentinel: &String, target: &PathBuf) {
+    info!(target: "files", "populating {target:?} from {source:?} with sentinel {sentinel:?}");
+    let sentinel_src = source.join(sentinel);
+
+    if !sentinel_src.exists() {
+        info!(target: "files", "sentinel file ({0:?}) not found, skipping", &sentinel_src);
+        return;
+    }
+
+    let sentinel_data = match fs::read(&sentinel_src) {
+        Ok(d) if d.trim_ascii() == b"" => {
+            error!(target: "files", "found empty sentinel file ({0:?})", &sentinel_src);
+            return;
+        }
+        Ok(d) => d,
+        Err(e) => {
+            error!(target: "files", "failed to read sentinel file: {e}");
+            return;
+        }
+    };
+
+    let sentinel_tgt = target.join(sentinel);
+    if sentinel_data == fs::read(&sentinel_tgt).unwrap_or_default() {
```

The use of `.unwrap_or_default()` feels a little unclean here as it's being used to ignore the error case and create a new string. This check also relies on the condition that valid `sentinel_data != ""` in order to work, and although this may always be the case, it's an extra precondition which must be kept in mind. I think instead, it could be clearer to use something like this to explicitly ignore the error and then check the content if successfully read

```rust
fs::read(&sentinel_tgt).is_ok_and(|content| content == sentinel_data)
```